crates/aptos-genesis/src/config.rs

```diff
@@ -344,3 +354,217 @@ pub struct StringOperatorConfiguration {
+        let mut accounts = HashSet::new();
+        let mut balances = vec![];
+        for (i, balance_entry) in balance_map.account_balances.iter().enumerate() {
+            let (account_address, balance) = balance_entry
+                .iter()
+                .next()
+                .ok_or_else(|| anyhow::anyhow!("No account in entry #{}", i))?;
+
+            if !accounts.insert(*account_address) {
+                return Err(anyhow::anyhow!(
+                    "An account was duplicated {} in the balances at entry #{}",
+                    account_address,
+                    i
+                ));
+            }
+
+            balances.push(AccountBalance {
+                account_address: *account_address,
+                balance: *balance,
+            });
+        }
+
+        Ok(balances)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct EmployeePoolMap {
+    pub inner: Vec<EmployeePoolConfig>,
+}
+
+impl Serialize for EmployeePoolMap {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.inner.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for EmployeePoolMap {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let inner = <Vec<EmployeePoolConfig>>::deserialize(deserializer)?;
+        Ok(EmployeePoolMap { inner })
+    }
+}
+
+impl TryFrom<EmployeePoolMap> for Vec<EmployeePool> {
+    type Error = anyhow::Error;
+
+    fn try_from(map: EmployeePoolMap) -> Result<Self, Self::Error> {
+        let mut employee_accounts = HashSet::new();
+        let mut pools = vec![];
+        for (i, pool) in map.inner.into_iter().enumerate() {
+            // Check for duplicate employee accounts
+            for (j, employee_account) in pool.accounts.iter().enumerate() {
+                if !employee_accounts.insert(*employee_account) {
+                    anyhow::bail!(
+                        "Employee account #{} {} duplicated in pool #{}",
+                        j,
+                        employee_account,
+                        i
+                    )
+                }
+            }
+
+            // Check vesting schedule adds up properly
+            let mut numerators = 0;
+            let denominator = pool.vesting_schedule_denominator;
+            let mut last_numerator = 0;
+            for numerator in pool.vesting_schedule_numerators.iter() {
+                numerators += *numerator;
+                last_numerator = *numerator;
+            }
+
+            if numerators > denominator {
+                anyhow::bail!(
+                    "Numerators {} add up over the denominator {} for pool #{}",
+                    numerators,
+                    denominator,
+                    i
+                )
+            } else if (denominator - numerators) % last_numerator != 0 {
```

Comment:
While we're at it should we check that the denominator isn't zero? (if we don't already)